lib/prism/ripper_compat.rb

```diff
@@ -142,31 +147,92 @@ def visit_call_node(node)
       if node.block
         block_val = visit(node.block)
 
-        return on_method_add_block(args_call_val, on_brace_block(nil, block_val))
+        return on_method_add_block(args_call_val, block_val)
       else
         return args_call_val
       end
     end
 
-    # Visit a BlockNode
+    # Visit a LocalVariableAndWriteNode.
+    def visit_local_variable_and_write_node(node)
+      visit_binary_op_assign(node)
+    end
+
+    # Visit a LocalVariableOrWriteNode.
+    def visit_local_variable_or_write_node(node)
+      visit_binary_op_assign(node)
+    end
+
+    # Visit nodes for +=, *=, -=, etc., called LocalVariableOperatorWriteNodes.
+    def visit_local_variable_operator_write_node(node)
+      visit_binary_op_assign(node, operator: node.operator.to_s + "=")
+    end
+
+    # Visit a LocalVariableReadNode.
+    def visit_local_variable_read_node(node)
+      bounds(node.location)
+      ident_val = on_ident(node.slice)
+
+      on_var_ref(ident_val)
+    end
+
+    # Visit a BlockNode.
     def visit_block_node(node)
-      if node.body.nil?
-        on_stmts_add(on_stmts_new, on_void_stmt)
-      else
-        visit(node.body)
-      end
+      params_val = node.parameters.nil? ? nil : visit(node.parameters)
+
+      body_val = node.body.nil? ? on_stmts_add(on_stmts_new, on_void_stmt) : visit(node.body)
+
+      on_brace_block(params_val, body_val)
+    end
+
+    # Visit a BlockParametersNode.
+    def visit_block_parameters_node(node)
+      on_block_var(visit(node.parameters), no_block_value)
+    end
+
+    # Visit a ParametersNode.
+    # This will require expanding as we support more kinds of parameters.
+    def visit_parameters_node(node)
+      #on_params(required, optional, nil, nil, nil, nil, nil)
+      on_params(node.requireds.map { |n| visit(n) }, nil, nil, nil, nil, nil, nil)
```

Comment:
Ahh this is the first place where we need `visit_all`. We should change the parent class to `Prism::Compiler` instead of `Prism::Visitor` so that we can use it.

```suggestion
      on_params(visit_all(node.requireds), nil, nil, nil, nil, nil, nil)
```

Reply:
Okay. I did this and changed the parent class.